utils/hct/hcttest.cmd

```diff
@@ -265,7 +265,20 @@ if "%TEST_CLEAN%"=="1" (
 )
 
 if "%TEST_MANUAL_FILE_CHECK%"=="1" (
-  set TEST_USE_LIT=0
+  echo %MANUAL_FILE_CHECK_PATH%|find "\HLSLFileCheck\" >nul
+  if errorlevel 1 (
```

Comment:
What's the intention here?  If `"\HLSLFileCheck\"` is *not* in the path given, it will use lit.py, otherwise it runs the legacy manual file-check path.  I don't understand why this is the intended behavior.

Reply:
It is to support hcttest file-check for both taef File check test and lit File check shell test.

Once we convert all taef File check tests to lit File check shell test, we can remove this just use llvm-lit directly.

Reply:
Ok, got it.  Maybe we need a comment explaining this.

Also, how are relative paths handled?  Currently, it would pass the relative path to TAEF, and the ManualFileCheck test would interpret a relative path as relative to the HLSLFileCheck directory.  This would change that behavior to use lit on a relative path from the current directory instead.  That sounds ok, unless the relative path is under HLSLFileCheck.  We could check the absolute version of the input path for HLSLFileCheck instead, which could catch this case, and we would pass that absolute version into ManualFileCheck.  This would change the relative path behavior to always be relative to the current working directory (which I think is better - I never used the other relative path behavior, and the previous behavior couldn't be used for lit otherwise).

Reply:
relative paths handled by use %~f2 instead of %~2.